ksp/src/main/kotlin/net/skyscanner/backpack/ksp/visitor/ComposeStoriesVisitor.kt

```diff
@@ -0,0 +1,42 @@
+package net.skyscanner.backpack.ksp.visitor
+
+import com.google.devtools.ksp.symbol.FileLocation
+import com.google.devtools.ksp.symbol.KSFunctionDeclaration
+import com.google.devtools.ksp.symbol.KSName
+import com.google.devtools.ksp.symbol.KSNode
+import com.google.devtools.ksp.visitor.KSDefaultVisitor
+import net.skyscanner.backpack.ksp.ComponentDefinition
+import net.skyscanner.backpack.ksp.ComposeStoryAnnotation
+import net.skyscanner.backpack.ksp.StoryDefinition
+import net.skyscanner.backpack.ksp.find
+import net.skyscanner.backpack.ksp.get
+
+object ComposeStoriesVisitor : KSDefaultVisitor<Map<KSName, ComponentDefinition>, StoryDefinition?>() {
+
+  override fun visitFunctionDeclaration(function: KSFunctionDeclaration, data: Map<KSName, ComponentDefinition>): StoryDefinition? {
+    val annotation = function.annotations.find(ComposeStoryAnnotation)
+    val location = function.location
+
+    val componentAnnotation = function.annotations.firstNotNullOfOrNull {
+      it.annotationType.resolve().declaration.qualifiedName?.let(data::get)
+    }
+
+    return when {
+      annotation != null && location is FileLocation -> StoryDefinition(
+        component = componentAnnotation ?: error("No component definition is found!"),
+        name = annotation[ComposeStoryAnnotation.paramName],
+        isCompose = true,
+        isScreenshot = annotation[ComposeStoryAnnotation.paramScreenshot],
```

Comment:
we're not supporting screenshot bits yet, right?

Reply:
It's just easier to have it there, and transferring from spike branch was also easier